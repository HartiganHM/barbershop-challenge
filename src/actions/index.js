import getPhotos from '../helpers/getPhotos/getPhotos';
import saveInitialPhotos from '../helpers/saveInitialPhotos/saveInitialPhotos';
import getSavedPhotos from '../helpers/getSavedPhotos/getSavedPhotos';

export const populatePhotos = () => async dispatch => {
  let photos = getSavedPhotos();

  if (!photos) {
    photos = await getPhotos();
  }

  saveInitialPhotos(photos);
  dispatch(sendPhotosToStore(photos));
};

export const sendPhotosToStore = photos => ({
  type: 'PHOTOS_TO_STORE',
  photos
});

export const changePhotosView = selectedView => ({
  type: 'CHANGE_VIEW',
  selectedView
});

export const sendSelectedPhotoToStore = selectedPhoto => ({
  type: 'SET_SELECTED_PHOTO',
  selectedPhoto
});
