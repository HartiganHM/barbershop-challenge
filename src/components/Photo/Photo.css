@import url("https://fonts.googleapis.com/css?family=Open+Sans");
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
.Photo {
  display: flex;
  align-items: center; }
  .Photo .list-image {
    width: 24rem;
    margin: 2rem 0; }
  .Photo .photo-details-link {
    margin: 0 4rem;
    color: #2F80ED;
    font-size: 3.2rem;
    font-weight: 600;
    line-height: 137.5%;
    word-break: break-all; }

@media screen and (max-width: 1260px) {
  .Photo .grid-image {
    width: 27rem; }
  .Photo .photo-details-link {
    margin: 0 20px;
    font-size: 24px; } }

@media screen and (max-width: 615px) {
  .Photo .list-image {
    width: 80px; }
  .Photo .photo-details-link {
    font-size: 16px; } }
