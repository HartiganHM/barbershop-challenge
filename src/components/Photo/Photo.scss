@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';

.Photo {
  display: flex;
  align-items: center;
  .list-image {
    width: 24rem;
    margin: 2rem 0;
  }
  .photo-details-link {
    margin: 0 4rem;
    color: $blue-midtone;
    font-size: 3.2rem;
    font-weight: 600;
    line-height: 137.5%;
    word-break: break-all;
  }
}

@media screen and (max-width: 1260px) {
  .Photo {
    .grid-image {
      width: 27rem;
    }
    .photo-details-link {
      margin: 0 20px;
      font-size: 24px;
    }
  }
}

@media screen and (max-width: 615px) {
  .Photo {
    .list-image {
      width: 80px;
    }
    .photo-details-link {
      font-size: 16px;
    }
  }
}