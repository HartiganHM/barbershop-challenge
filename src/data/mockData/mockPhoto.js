/*eslint-disable camelcase*/

const mockPhoto = {
  categories: [],
  color: '#DF8A65',
  created_at: '2017-07-20T02:21:08-04:00',
  current_user_collections: [],
  description:
    'A highway road through rocky cliffs at Valley of Fire State Park',
  height: 5472,
  id: 'tMzCrBkM99Y',
  liked_by_user: false,
  likes: 2306,
  links: {
    download: 'https://unsplash.com/photos/tMzCrBkM99Y/download',
    download_location: 'https://api.unsplash.com/photos/tMzCrBkM99Y/download',
    html: 'https://unsplash.com/photos/tMzCrBkM99Y',
    self: 'https://api.unsplash.com/photos/tMzCrBkM99Y'
  },
  sponsored: false,
  updated_at: '2017-11-01T15:04:31-04:00',
  urls: {
    full:
      'https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=b09cf725577ce730f9f8323a6a17037d',
    raw:
      'https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=8da1e8ea486b7798563909d68f9f0978',
    regular:
      'https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=b29adb2b9778dbc6782f49370e3e3bcf',
    small:
      'https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95',
    thumb:
      'https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=e713f9e40adacbd68dac2789f33daecd'
  },
  user: {
    bio: `Top 100 Most Downloaded Photos of the Year 2017 |
    Editorâ€™s Choice Top 100 Photos of the Year 2017 |
    Follow my adventures on Instagram: @jakeblucker`,
    first_name: 'Jake',
    id: 'Ev2n9fB9hXc',
    instagram_username: 'jakeblucker',
    last_name: 'Blucker',
    links: {
      followers: 'https://api.unsplash.com/users/jakeblucker/followers',
      following: 'https://api.unsplash.com/users/jakeblucker/following',
      html: 'https://unsplash.com/@jakeblucker',
      likes: 'https://api.unsplash.com/users/jakeblucker/likes',
      photos: 'https://api.unsplash.com/users/jakeblucker/photos',
      portfolio: 'https://api.unsplash.com/users/jakeblucker/portfolio',
      self: 'https://api.unsplash.com/users/jakeblucker'
    },
    location: null,
    name: 'Jake Blicker',
    portfolio_url: null,
    profile_image: {
      large:
        'https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128&s=ead1dab7076e5e3eea6b696f2fc25580',
      medium:
        'https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=c354801257e2365fda630ae3fea7a155',
      small:
        'https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32&s=f4f2a9a39c5fa6e03de2ee43829b4dda'
    },
    total_collections: 0,
    total_likes: 33,
    total_photos: 11,
    twitter_username: 'JakeBlucker',
    updated_at: '2018-03-11T03:35:48-04:00',
    username: 'jakeblucker'
  },
  width: 3648
};

export default mockPhoto;
