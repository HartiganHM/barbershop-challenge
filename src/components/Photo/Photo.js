import React from 'react';
import './Photo.css';

const Photo = () => {
  return (
    <div className="Photo">
      <h1>Soi una Photo</h1>
    </div>
  )
}

export default Photo;