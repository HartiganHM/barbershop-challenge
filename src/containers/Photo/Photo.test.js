import React from 'react';
import { shallow } from 'enzyme';
import { Photo, mapDispatchToProps } from './Photo';
import mockPhoto from '../../data/mockData/mockPhoto';
import * as actions from '../../actions';

describe('Photo tests', () => {

});

describe('mapDispatchToProps tests', () => {
  it('Should call dispatch when sendSelectedPhotoToStore is called', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.sendSelectedPhotoToStore();
    expect(mockDispatch).toHaveBeenCalled();
  });
});