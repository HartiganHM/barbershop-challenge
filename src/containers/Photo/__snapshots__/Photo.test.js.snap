// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentView="List"
    photoData={
      Object {
        "categories": Array [],
        "color": "#DF8A65",
        "created_at": "2017-07-20T02:21:08-04:00",
        "current_user_collections": Array [],
        "description": "A highway road through rocky cliffs at Valley of Fire State Park",
        "height": 5472,
        "id": "tMzCrBkM99Y",
        "liked_by_user": false,
        "likes": 2306,
        "links": Object {
          "download": "https://unsplash.com/photos/tMzCrBkM99Y/download",
          "download_location": "https://api.unsplash.com/photos/tMzCrBkM99Y/download",
          "html": "https://unsplash.com/photos/tMzCrBkM99Y",
          "self": "https://api.unsplash.com/photos/tMzCrBkM99Y",
        },
        "sponsored": false,
        "updated_at": "2017-11-01T15:04:31-04:00",
        "urls": Object {
          "full": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=b09cf725577ce730f9f8323a6a17037d",
          "raw": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=8da1e8ea486b7798563909d68f9f0978",
          "regular": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=b29adb2b9778dbc6782f49370e3e3bcf",
          "small": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95",
          "thumb": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=e713f9e40adacbd68dac2789f33daecd",
        },
        "user": Object {
          "bio": "Top 100 Most Downloaded Photos of the Year 2017 |
    Editorâ€™s Choice Top 100 Photos of the Year 2017 |
    Follow my adventures on Instagram: @jakeblucker",
          "first_name": "Jake",
          "id": "Ev2n9fB9hXc",
          "instagram_username": "jakeblucker",
          "last_name": "Blucker",
          "links": Object {
            "followers": "https://api.unsplash.com/users/jakeblucker/followers",
            "following": "https://api.unsplash.com/users/jakeblucker/following",
            "html": "https://unsplash.com/@jakeblucker",
            "likes": "https://api.unsplash.com/users/jakeblucker/likes",
            "photos": "https://api.unsplash.com/users/jakeblucker/photos",
            "portfolio": "https://api.unsplash.com/users/jakeblucker/portfolio",
            "self": "https://api.unsplash.com/users/jakeblucker",
          },
          "location": null,
          "name": "Jake Blicker",
          "portfolio_url": null,
          "profile_image": Object {
            "large": "https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128&s=ead1dab7076e5e3eea6b696f2fc25580",
            "medium": "https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64&s=c354801257e2365fda630ae3fea7a155",
            "small": "https://images.unsplash.com/profile-1516312729390-1fb554b25f8b?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32&s=f4f2a9a39c5fa6e03de2ee43829b4dda",
          },
          "total_collections": 0,
          "total_likes": 33,
          "total_photos": 11,
          "twitter_username": "JakeBlucker",
          "updated_at": "2018-03-11T03:35:48-04:00",
          "username": "jakeblucker",
        },
        "width": 3648,
      }
    }
    sendSelectedPhotoToStore={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="A highway road through rocky cliffs at Valley of Fire State Park"
          className="list-image"
          src="https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95"
        />,
        <h4
          className="photo-details-link"
        >
          https://unsplash.com/photos/tMzCrBkM99Y
        </h4>,
      ],
      "className": "Photo",
      "onClick": [Function],
      "replace": false,
      "to": "/details/tMzCrBkM99Y",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "A highway road through rocky cliffs at Valley of Fire State Park",
          "className": "list-image",
          "src": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "https://unsplash.com/photos/tMzCrBkM99Y",
          "className": "photo-details-link",
        },
        "ref": null,
        "rendered": "https://unsplash.com/photos/tMzCrBkM99Y",
        "type": "h4",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="A highway road through rocky cliffs at Valley of Fire State Park"
            className="list-image"
            src="https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95"
          />,
          <h4
            className="photo-details-link"
          >
            https://unsplash.com/photos/tMzCrBkM99Y
          </h4>,
        ],
        "className": "Photo",
        "onClick": [Function],
        "replace": false,
        "to": "/details/tMzCrBkM99Y",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "A highway road through rocky cliffs at Valley of Fire State Park",
            "className": "list-image",
            "src": "https://images.unsplash.com/photo-1500530855697-b586d89ba3ee?ixlib=rb-0.3.5&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjIyNDUyfQ&s=bb51a2166a2929a330177532df56bd95",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "https://unsplash.com/photos/tMzCrBkM99Y",
            "className": "photo-details-link",
          },
          "ref": null,
          "rendered": "https://unsplash.com/photos/tMzCrBkM99Y",
          "type": "h4",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
